app/src/main/java/com/example/mobilelibrary/util/TextTranslator.java

```diff
@@ -0,0 +1,78 @@
+package com.example.mobilelibrary.util;
+
+import androidx.annotation.NonNull;
+import android.util.Log;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.ml.common.modeldownload.FirebaseModelDownloadConditions;
+import com.google.firebase.ml.naturallanguage.FirebaseNaturalLanguage;
+import com.google.firebase.ml.naturallanguage.translate.FirebaseTranslateLanguage;
+import com.google.firebase.ml.naturallanguage.translate.FirebaseTranslator;
+import com.google.firebase.ml.naturallanguage.translate.FirebaseTranslatorOptions;
+
+public class TextTranslator {
+
+    private String translatedText;
+    private final String TAG = "TRANSLATE";
+
+    private FirebaseTranslator getFirebaseTranslator() {
+        FirebaseTranslatorOptions options = new FirebaseTranslatorOptions.Builder()
+                .setSourceLanguage(FirebaseTranslateLanguage.EN)
+                .setTargetLanguage(FirebaseTranslateLanguage.RO)
+                .build();
+        return FirebaseNaturalLanguage.getInstance().getTranslator(options);
+    }
+
+    public void translateText(String textToTranslate) {
+
+        FirebaseTranslator translator = getFirebaseTranslator();
+
+        FirebaseModelDownloadConditions conditions = new FirebaseModelDownloadConditions.Builder()
+            .requireWifi()
+            .build();
+
+        translator.downloadModelIfNeeded(conditions)
```

Comment:
this should not be in this flow. there should be a separate activity where the user adds languages he wants. AddButtonOnClick -> SelectLanguage -> ConfirmButtonOnClick -> DownloadIfNeeded. and start by default with RO and EN, so call DownloadIfNeeded for RO and EN in MainActivity.OnCreate, or Application.OnCreate. so you could add the translator as a property of Application. check [this](https://guides.codepath.com/android/Understanding-the-Android-Application-Class)